app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java

```
@@ -906,6 +906,52 @@ public enum AppsmithError {
             "Invalid application property configuration",
             ErrorType.INTERNAL_ERROR,
             null),
+
+    INVALID_SMTP_CONFIGURATION(
+            400,
+            AppsmithErrorCode.INVALID_SMTP_CONFIGURATION.getCode(),
+            "Your SMTP configuration is invalid. Please configure SMTP to enable to proceed",
+            AppsmithErrorAction.DEFAULT,
+            "Invalid SMTP configuration",
+            ErrorType.INTERNAL_ERROR,
+            null),
+
+    USER_ALREADY_VERIFIED(
+            400,
+            AppsmithErrorCode.USER_EMAIL_ALREADY_VERIFIED.getCode(),
+            "This email has already been verified",
+            AppsmithErrorAction.DEFAULT,
+            "User email already verified",
+            ErrorType.BAD_REQUEST,
+            null),
+
+    EMAIL_VERIFICATION_TOKEN_EXPIRED(
+            498,
+            AppsmithErrorCode.EMAIL_VERIFICATION_TOKEN_EXPIRED.getCode(),
+            "Your email verification link has expired. Please re-trigger email verification",
+            AppsmithErrorAction.DEFAULT,
+            "Email verification token expired",
+            ErrorType.INTERNAL_ERROR,
+            null),
+
+    TENANT_EMAIL_VERIFICATION_NOT_ENABLED(
```

In what case we want to throw this exception? If not required let's remove this

---

It's being used in the sending verification email method, which returns an error in case the email verification is not enabled but still trying to invoke this method.
[Reference](https://github.com/appsmithorg/appsmith/blob/3732273cccafd7ba7bdd6885c3eaea1dd8471158/app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/UserServiceCEImpl.java#L924)